test/testrunner.py

```diff
@@ -120,21 +120,19 @@ def pull_dlc_images(images):
         run(f"docker pull {image}", hide="out")
 
 
-def setup_eks_clusters(dlc_images):
+def setup_eks_clusters(framework_name):
     frameworks = {"tensorflow": "tf", "pytorch": "pt", "mxnet": "mx"}
-    frameworks_in_images = [framework for framework in frameworks.keys() if framework in dlc_images]
-    if len(frameworks_in_images) != 1:
-        raise ValueError(
-            f"All images in dlc_images must be of a single framework for EKS tests.\n"
-            f"Instead seeing {frameworks_in_images} frameworks."
-        )
-    long_name = frameworks_in_images[0]
+    long_name = framework_name
     short_name = frameworks[long_name]
-    num_nodes = 2 if is_pr_context() else 3 if long_name != "pytorch" else 4
+    num_nodes = 1 if is_pr_context() else 3 if long_name != "pytorch" else 4
     cluster_name = f"dlc-{short_name}-cluster-" \
                    f"{os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')}-{random.randint(1, 10000)}"
-    eks_utils.create_eks_cluster(cluster_name, "gpu", num_nodes, "p3.16xlarge", "pytest.pem")
-    eks_utils.eks_setup(long_name, cluster_name)
+    try:
+        eks_utils.create_eks_cluster(cluster_name, "gpu", num_nodes, "p3.16xlarge", "pytest.pem")
+        eks_utils.eks_setup(long_name, cluster_name)
+    except Exception as e:
+        eks_utils.delete_eks_cluster(cluster_name)
+        raise Exception(e)
```

Comment:
This will now raise exception type "Exception"

Consider the following two examples:

In the first, we do not overwrite the exception as "Exception" type and get an assertion error.
```
In [2]: try:
   ...:     assert 1==2, "not true"
   ...: except:
   ...:     print('test')
   ...:     raise
   ...:
   ...:
test
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-2-52478dbb11bf> in <module>
      1 try:
----> 2     assert 1==2, "not true"
      3 except:
      4     print('test')
      5     raise

AssertionError: not true
```

Here, we raise "Exception(e)" and ultimately surface a redundant "Exception" error
```
In [3]: try:
   ...:     assert 1==2, "not true"
   ...: except Exception as e:
   ...:     print('test')
   ...:     raise Exception(e)
   ...:
   ...:
   ...:
test
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-3-698ec5fdd6d8> in <module>
      1 try:
----> 2     assert 1==2, "not true"
      3 except Exception as e:

AssertionError: not true

During handling of the above exception, another exception occurred:

Exception                                 Traceback (most recent call last)
<ipython-input-3-698ec5fdd6d8> in <module>
      3 except Exception as e:
      4     print('test')
----> 5     raise Exception(e)
      6
      7

Exception: not true
```

Reply:
Changing this. How about
```
try:
    assert False, "not true"
except Exception:
    print("Running cleanup tasks")
    raise
```
because it looks like an
```
except:
```
statement goes against PEP recommendations, but
```
except Exception:
```
doesn't, even though we do end up not specifically using the Exception information.

Reply:
Sure that's fine. You can also do this if you want to be a little more clear.

```
In [2]: try:
   ...:     assert 1==2, "not true"
   ...: except Exception as e:
   ...:     print('test')
   ...:     raise(e)
   ...:
   ...:
test
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-2-d91d2ae79f44> in <module>
      3 except Exception as e:
      4     print('test')
----> 5     raise(e)
      6
      7

<ipython-input-2-d91d2ae79f44> in <module>
      1 try:
----> 2     assert 1==2, "not true"
      3 except Exception as e:
      4     print('test')
      5     raise(e)

AssertionError: not true
```

Reply:
I tried this on my system - It caused two instances of the same Exception, one from the line where it originally came, and the other where it was raised again.

That's why I went with the approach I used.